src/Tools/dotnet-dump/Analyzer.cs

```diff
@@ -22,56 +22,76 @@ namespace Microsoft.Diagnostics.Tools.Dump
 {
     public class Analyzer : IHost
     {
-        private readonly ServiceProvider _serviceProvider;
+        private readonly ServiceManager _serviceManager;
+        private readonly IServiceContainer _serviceContainer;
         private readonly ConsoleService _consoleService;
         private readonly FileLoggingConsoleService _fileLoggingConsoleService;
         private readonly CommandService _commandService;
         private readonly SymbolService _symbolService;
         private readonly ContextService _contextService;
+        private readonly List<ITarget> _targets = new();
         private int _targetIdFactory;
-        private Target _target;
 
         public Analyzer()
         {
             DiagnosticLoggingService.Initialize();
 
-            _serviceProvider = new ServiceProvider();
+            _serviceManager = new ServiceManager();
+            _serviceContainer = _serviceManager.CreateServiceContainer(ServiceScope.Global);
+            _serviceContainer.AddService<IServiceManager>(_serviceManager);
+            _serviceContainer.AddService<IHost>(this);
+
             _consoleService = new ConsoleService();
-            _fileLoggingConsoleService  = new FileLoggingConsoleService(_consoleService);
+            _fileLoggingConsoleService = new FileLoggingConsoleService(_consoleService);
+            _serviceContainer.AddService<IConsoleService>(_fileLoggingConsoleService);
+            _serviceContainer.AddService<IConsoleFileLoggingService>(_fileLoggingConsoleService);
+
+            DiagnosticLoggingService.Instance.SetConsole(_fileLoggingConsoleService, _fileLoggingConsoleService);
```

Comment:
Any reason to maintain both the service initialization and the global static? The service never gets used

Reply:
IDiagnosticLoggingService does get used by the logging command.

The reason for the static Initialize and Instance property to get logging (to file via the env var) setup as soon as possible.